### ISSUE TITLE
test_windows_get_args_for_reloading fails when run with python -m pytest

### ISSUE THREAD

**Issue by frenzymadness** [pallets/werkzeug#1695]
Hello.

`test_windows_get_args_for_reloading` fails when the test suite is executed with `python3 -m pytest` instead of just `pytest` which I think is incorrect.

```
$ python3 -m pytest -vv test_serving.py
…
================================== FAILURES ===================================
_____________________ test_windows_get_args_for_reloading _____________________

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f105e0ecd60>
tmpdir = local('/tmp/pytest-of-lbalhar/pytest-4/test_windows_get_args_for_relo0')

    def test_windows_get_args_for_reloading(monkeypatch, tmpdir):
        test_py_exe = r"C:\Users\test\AppData\Local\Programs\Python\Python36\python.exe"
        monkeypatch.setattr(os, "name", "nt")
        monkeypatch.setattr(sys, "executable", test_py_exe)
        test_exe = tmpdir.mkdir("test").join("test.exe")
        monkeypatch.setattr(sys, "argv", [test_exe.strpath, "run"])
        rv = _reloader._get_args_for_reloading()
>       assert rv == [test_exe.strpath, "run"]
E       AssertionError: assert ['C:\\Users\\test\\AppData\\Local\\Programs\\Python\\Python36\\python.exe',\n '-m',\n '/tmp/pytest-of-lbalhar/pytest-4/test_windows_get_args_for_relo0/test/test.exe',\n 'run'] == ['/tmp/pytest-of-lbalhar/pytest-4/test_windows_get_args_for_relo0/test/test.exe',\n 'run']
E         At index 0 diff: 'C:\\Users\\test\\AppData\\Local\\Programs\\Python\\Python36\\python.exe' != '/tmp/pytest-of-lbalhar/pytest-4/test_windows_get_args_for_relo0/test/test.exe'
E         Left contains 2 more items, first extra item: '/tmp/pytest-of-lbalhar/pytest-4/test_windows_get_args_for_relo0/test/test.exe'
E         Full diff:
E           [
E         -  'C:\\Users\\test\\AppData\\Local\\Programs\\Python\\Python36\\python.exe',
E         -  '-m',
E            '/tmp/pytest-of-lbalhar/pytest-4/test_windows_get_args_for_relo0/test/test.exe',
E            'run',
E           ]

test_serving.py:305: AssertionError
```
but
```
$ pytest test_serving.py 
============================= test session starts =============================
platform linux -- Python 3.8.1, pytest-5.3.2, py-1.8.1, pluggy-0.13.1
rootdir: /home/lbalhar/Fedora/packages/python-werkzeug/Werkzeug-0.16.0, inifile: setup.cfg
collected 21 items                                                            

test_serving.py ssssssssss..sssssssss                                   [100%]
```

`python3 -m pytest` might be a preferred way in some cases when, for example, `/usr/bin/pytest` might still point to Python 2 version of Pytest.

**Comment by davidism** [comment]
I can't reproduce this with the latest master. Probably fixed in #1690.

**Comment by frenzymadness** [comment]
Yes, you are right. Thank you.
